docs/example-terraform.md

```diff
@@ -0,0 +1,187 @@
+  [if name_in_resoures("iam") && "serviceaccounts" in p.resources.iam then "iam_service_account.tf"]: iam_service_account,
+...
+  [if name_in_resoures("pubsub") then "pubsub.tf"]: pubsub,
+  [if name_in_resoures("storage") then "storage.tf"]: storage,
+}
+```
+
+Each Jsonnet file defines a resource and then it is imported. Jsonnet then filters through all the inventory parameters to find specific keys that have been defined. Let's take for example the cloud build resource: 
+
+```json5
+local cloudbuild = import "cloudbuild.jsonnet";
+...
+{
+  "output.tf": output,
+  "provider.tf": provider,
+  [if name_in_resoures("cloudbuild") then "cloudbuild.tf"]: cloudbuild,
+...
+}
+```  
+
+Assuming that one of the configuration files for a specific environment has the parameter key `cloudbuild` set. 
+These parameters will then be interpreted by the `cloudbuild.jsonnet` template. 
+A file named `cloudbuild.tf.json` will then be compiled using the parameters associated with the `cloudbuild` parameter key. 
+
+It is important to understand that once you have a deeper understanding of Kapitan's capabilities, you can organize these files to a style & logic suitable for your organization.
+
+### templates, docs, scripts
+
+Jinja2 is used to generate documentation that sources information from kapitan's inventory. This enables the ability to have dynamic documentation based on your infrastructure configuration changes.
+In `templates/terraform` you will find `README.md.j2`. This is used to generate a `README.md` template to be utilized by terraform's output module.
+
+The following is what generates the documentation:
+
+```json5
+  data: {
+    template_file: {
+      readme: {
+        template: kap.jinja2_template("templates/terraform/README.md.j2", inv),
+      },
+    },
+  },
+
+  output: {
+    "README.md": {
+      value: "${data.template_file.readme.rendered}",
+      sensitive: true,
+    },
+  },
+```
+
+The function `kap.jinja2_template()` (imported from `kapitan.libjsonnet`) is used to convert and interpreter the `README.md.j2` file into a raw string using the inventory for evaluation logic.
+
+Based on the various parameters jinja2 decides which sections of the readme should be included. 
+When terraform runs it will use the output module to generate your desired `README.md` using information from terraform's state.
+
+Scripts are located in the `scripts` directory. They are compiled using jinja2 as the templating language. An example is as follows:
+
+```shell
+export TF_DATA_DIR=$(realpath -m ${DIR}/../../../.TF_DATA_DIR/{{inventory.parameters.name}}) # Folder for TF initialization (preferable outside of compiled)
+export OUTPUT_DIR=$(realpath -m ${DIR}/../../../output/{{inventory.parameters.name}}) # Folder for storing output files (preferable outside of compiled)
+```
+
+It is good practice to utilize this method to improve integration with various CLI based tools. Scripts help to ensure terraform &
```

Comment:
All this looks great. :) Appreciate the effort!
One last comment is if you could replace `&` with `and` as it makes searching codebases/documentation more intuitive?

Reply:
I never considered that before. Good point :D